File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
# import heapq
from heapq import heapify, heappop, heappush
def solution(scoville, K):
    answer = 0
    heapify(scoville)
    while scoville[0] < K:
        if len(scoville) < 2:
            break
        a = heappop(scoville)
        b = heappop(scoville)
        heappush(scoville, a + (b * 2))
        answer += 1
    if (len(scoville) <= 1 and scoville[0] < K):
        return -1
        
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {

    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> queue = new PriorityQueue<>();
        for (Integer i : scoville) {
            queue.add(i);
        }        
        while (!(queue.peek() >= K)) {
            if (queue.size() == 1) {
                return -1;
            }
            int x = queue.poll();
            int y = queue.poll();
            int temp = x + (y * 2);
            queue.add(temp);
            answer ++;
        }
        
        return answer;
    }
}